app/assets/v2/js/pages/bounty_details.js

```diff
@@ -719,6 +719,23 @@ const is_current_user_interested = function(result) {
   return !!(result.interested || []).find(interest => interest.profile.handle == document.contxt.github_handle);
 };
 
+const is_current_user_approved = function(result) {
+  const needs_approval = result['permission_type'] === 'approval';
+  const interested = result.interested || [];
+  let len = interested.length;
+
+  while (len--) {
+    const interest = interested[len];
+    const handle = interest.profile ? interest.profile.handle : '';
+
+    if (handle && handle === document.contxt.github_handle) {
+      return needs_approval ? interest.pending === false : true;
```

Comment:
I think just checking interest.pending isn't enough. We also need to check the `interest.status` and the `interest.acceptance_date` to  ensure that it is actually accepted and not some other state. cc: @mbeacom

Reply:
@SaptakS I will grep through the codebase and take a look at it 👍

Reply:
Seems like we rely on this too at
https://github.com/gitcoinco/web/blob/master/app/dashboard/models.py#L557
and we mutate 'pending' in
https://github.com/gitcoinco/web/blob/master/app/dashboard/views.py#L859

Reply:
and
https://github.com/gitcoinco/web/blob/master/app/marketing/management/commands/pending_start_work_actions.py#L28

Reply:
@pinkiebell we should update that too!
@SaptakS does have a valid point -> those extra checks ensures we don't display it in the right states.

Any reason you would be against it  ?
PS: could you fix resolve the conflicts too

Reply:
Sorry for late reply here. I would say the presence of `interest.acceptance_date` in the table would ensure. However, I am okay with resolving this thing later all over the codebase if that is more easier since I think we should get this shipped already. :)